cpp/src/copying/contiguous_split.cu

```diff
@@ -226,12 +236,21 @@ __device__ void copy_buffer(uint8_t* __restrict__ dst,
       }
     }
 
-    // if we're performing a value shift (offsets), or a bit shift (validity) the # of bytes and
-    // alignment must be a multiple of 4. value shifting and bit shifting are mutually exclusive
-    // and will never both be true at the same time.
-    if (value_shift || bit_shift) {
+    // if we're performing a value shift (offsets)
+    if constexpr (is_offsets) {
+      std::size_t idx = (num_bytes - remainder) / sizeof(offset_type);
+      while (remainder) {
+        reinterpret_cast<offset_type*>(dst)[idx] =
+          (reinterpret_cast<offset_type const*>(src)[idx] - value_shift);
+        idx++;
+        remainder -= sizeof(offset_type);
```

Comment:
Can this underflow?

Reply:
Do we need a test with an odd number of elements?